### PR TITLE
feat(pubsub)!: Use new generated client classes

### PR DESCRIPTION
**Pull request by quartzmo**
closes: #6488

### CODE REVIEW

File: google-cloud-pubsub/lib/google/cloud/pubsub/service.rb [hunk 1]
```diff
@@ -38,65 +36,58 @@ class Service
...
 
-        def channel
-          require "grpc"
-          GRPC::Core::Channel.new host, chan_args, chan_creds
-        end
-
-        def chan_args
-          { "grpc.max_send_message_length"           => -1,
-            "grpc.max_receive_message_length"        => -1,
-            "grpc.keepalive_time_ms"                 => 300_000,
```
Comment (dazuma) [inline google-cloud-pubsub/lib/google/cloud/pubsub/service.rb:59]: Are these grpc args necessary for pubsub? If so, we probably need to set them somewhere else.

Comment (quartzmo) [reply google-cloud-pubsub/lib/google/cloud/pubsub/service.rb:59]: Ah, thanks!

File: google-cloud-pubsub/lib/google/cloud/pubsub/service.rb [hunk 2]
```diff
@@ -38,65 +36,58 @@ class Service
...
-            "grpc.service_config_disable_resolution" => 1 }
```
Comment (quartzmo) [inline google-cloud-pubsub/lib/google/cloud/pubsub/service.rb:60]: @dazuma Do we still need to set this to disable the gRPC service config? See PR #3443 and googleapis/gapic-generator#2778.

Comment (dazuma) [reply google-cloud-pubsub/lib/google/cloud/pubsub/service.rb:60]: Yes we still need to set this. (Indeed it is set by default: https://github.com/googleapis/google-cloud-ruby/blob/master/google-cloud-pubsub-v1/lib/google/cloud/pubsub/v1/publisher/client.rb#L973)

Comment (quartzmo) [reply google-cloud-pubsub/lib/google/cloud/pubsub/service.rb:60]: Right! OK, I'll just set the other 3 channel args that I mistakenly removed in the same Config default value.

Comment (quartzmo) [reply google-cloud-pubsub/lib/google/cloud/pubsub/service.rb:60]: Should we set `max_send_message_length`, `max_receive_message_length` and `keepalive_time_ms` in google-cloud-pubsub-v1 somehow, with the idea being that all users of the pubsub-v1 client need these settings?

Comment (dazuma) [reply google-cloud-pubsub/lib/google/cloud/pubsub/service.rb:60]: Hmm. That's a good point. If these should be applied to all users of pubsub-v1, then they really should be set somehow in the generated client.

I don't think this requires a generator change. You can update the default config values in a partial gapic helper.

**Comment by quartzmo** [comment]
Thanks! Will do.

File: google-cloud-pubsub/lib/google/cloud/pubsub/service.rb (hunk 2 above)
Comment (quartzmo) [reply google-cloud-pubsub/lib/google/cloud/pubsub/service.rb:60]: This is fixed in #7054

File: google-cloud-pubsub/.rubocop.yml [hunk 3]
```diff
@@ -5,8 +5,6 @@ AllCops:
   Exclude:
     - "acceptance/**/*"
     - "lib/google/pubsub/**/*"
```
Comment (dazuma) [inline google-cloud-pubsub/.rubocop.yml:7]: This one can be removed as well.

File: google-cloud-pubsub/.rubocop.yml [hunk 4]
```diff
@@ -5,8 +5,6 @@ AllCops:
...
-    - "lib/google/cloud/pubsub/v1.rb"
-    - "lib/google/cloud/pubsub/v1/**/*"
     - "lib/google/iam/**/*"
```
Comment (dazuma) [inline google-cloud-pubsub/.rubocop.yml:8]: I think this one is obsolete also.

**Comment by dazuma** [comment]
Can you add an alias `Google::Pubsub` (note lower-case "s") --> `Google::Cloud::PubSub` (upper-case "S")? This was defined in the old generated code (via a synth hack) for backward compatibility with old usage from a year or more ago, but is not part of the microgenerator output.

File: google-cloud-pubsub/lib/google/cloud/pubsub/subscription.rb [hunk 5]
```diff
@@ -346,7 +346,7 @@ def expires_in
         #   to unset.
         #
         def expires_in= ttl
-          new_expiration_policy = Google::Pubsub::V1::ExpirationPolicy.new ttl: Convert.number_to_duration(ttl)
+          new_expiration_policy = Google::Cloud::Pubsub::V1::ExpirationPolicy.new ttl: Convert.number_to_duration(ttl)
```
Comment (dazuma) [inline google-cloud-pubsub/lib/google/cloud/pubsub/subscription.rb:349]: s/Pubsub/PubSub/

(`Google::Cloud::Pubsub` is an alias of `Google::Cloud::PubSub` right now, so this isn't strictly necessary, but better to clean it up.)

**Comment by quartzmo** [comment]
> Can you add an alias Google::Pubsub (note lower-case "s") --> Google::Cloud::PubSub

@dazuma Where do I need to add it? It's still in the handwritten code, for example [pubsub.rb](https://github.com/googleapis/google-cloud-ruby/blob/fdabf9c935aee4e4e274eb9e4a9a6c8c94194ec7/google-cloud-pubsub/lib/google/cloud/pubsub.rb#L166).

**Comment by dazuma** [comment]
> > Can you add an alias Google::Pubsub (note lower-case "s") --> Google::Cloud::PubSub
> 
> @dazuma Where do I need to add it? It's still in the handwritten code, for example [pubsub.rb](https://github.com/googleapis/google-cloud-ruby/blob/fdabf9c935aee4e4e274eb9e4a9a6c8c94194ec7/google-cloud-pubsub/lib/google/cloud/pubsub.rb#L166).

That one is `Google::Cloud::Pubsub` -> `Google::Cloud::PubSub`. We also need `Google::Pubsub` -> `Google::Cloud::PubSub` (replacing [this line](https://github.com/googleapis/google-cloud-ruby/blob/master/google-cloud-pubsub/lib/google/pubsub/v1/pubsub_pb.rb#L222)). Adding it in the same file would be fine.

**Comment by quartzmo** [comment]
I'm going to hold off on merging this for a couple days in case #6493 is ready to be merged.

**Comment by dazuma** [comment]
Looks like there are a couple of mocks that need updating.

**Comment by quartzmo** [comment]
> Looks like there are a couple of mocks that need updating.

Thanks, I missed the need to run google-cloud-storage CI locally. Good thing kokoro didn't!

Just pushed a commit to fix these as well as an issue with storage acceptance tests.

Also, I searched for usages of the storage gem in other packages, and got these results. Do you think any of these might require updates in this PR?

```
google-cloud-ruby • google-cloud-asset/samples/Gemfile:
  25:   gem "google-cloud-pubsub", group: :test, path: "../../google-cloud-pubsub"
  33:   gem "google-cloud-pubsub", group: :test

google-cloud-ruby • google-cloud-container_analysis/samples/Gemfile:
  25: gem "google-cloud-pubsub", "~> 1.7"

google-cloud-ruby • integration/rails4_app/Gemfile:
  59: gem "google-cloud-pubsub", path: "../../google-cloud-pubsub"

google-cloud-ruby • integration/rails5_app/Gemfile:
  62: gem "google-cloud-pubsub", path: "../../google-cloud-pubsub"

google-cloud-ruby • integration/sinatra2_app/Gemfile:
  16: gem "google-cloud-pubsub", path: "../../google-cloud-pubsub"
```

**Comment by dazuma** [comment]
I don't think any of the other usages (which are generally in samples) try to mock pubsub, so it's probably fine.

**Pull request merged**